src/ingest-pipeline/airflow/dags/rebuild_multiple_metadata.py

```diff
@@ -0,0 +1,115 @@
+from pprint import pprint
+
+from airflow.operators.python import PythonOperator
+from airflow.operators.trigger_dagrun import TriggerDagRunOperator
+from airflow.configuration import conf as airflow_conf
+from datetime import datetime
+from airflow import DAG
+
+from utils import (
+    get_queue_resource,
+    get_preserve_scratch_resource,
+    get_soft_data,
+    get_tmp_dir_path,
+    encrypt_tok,
+    pythonop_get_dataset_state,
+)
+
+
+def get_uuid_for_error(**kwargs):
+    """
+    Return the uuid for the derived dataset if it exists, and of the parent dataset otherwise.
+    """
+    return None
+
+
+default_args = {
+    'start_date': datetime(2019, 1, 1),
+}
+
+with DAG('rebuild_multiple_metadata',
+           schedule_interval=None,
+           is_paused_upon_creation=False,
+           default_args=default_args,
+           user_defined_macros={
+               'tmp_dir_path': get_tmp_dir_path,
+               'preserve_scratch': get_preserve_scratch_resource('rebuild_metadata')
+           }) as dag:
+
+    def build_dataset_lists(**kwargs):
+        kwargs['dag_run'].conf['primary_datasets'] = []
+        kwargs['dag_run'].conf['processed_datasets'] = []
+
+        print('dag_run conf follows:')
+        pprint(kwargs['dag_run'].conf)
+        for uuid in kwargs['dag_run'].conf['uuids']:
+            soft_data = get_soft_data(uuid, **kwargs)
+
+            # If we got nothing back from soft_data, then let's try to determine using entity_api
+            if soft_data:
+                if soft_data.get('primary'):
+                    kwargs['dag_run'].conf['primary_datasets'].append(uuid)
+                else:
+                    kwargs['dag_run'].conf['processed_datasets'].append(uuid)
+            else:
+                my_callable = lambda **kwargs: uuid
+                ds_rslt = pythonop_get_dataset_state(dataset_uuid_callable=my_callable, **kwargs)
```

Comment:
`# Nitpicky`
Could just pass the lambda as the arg to dataset_uuid_callable since it's only used the once?
`ds_rslt = pythonop_get_dataset_state(dataset_uuid_callable=lambda **kwargs: uuid, **kwargs)`